src/main/java/com/gzs/daos/inmemory/InMemoryDao.java

```diff
@@ -0,0 +1,40 @@
+package com.gzs.daos.inmemory;
+
+import com.gzs.model.DatabaseEntity;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Setter
+public class InMemoryDao<T extends DatabaseEntity> {
+
+    protected Map<Integer, T> dataMap;
+
+    public InMemoryDao() {
+        this.dataMap = new HashMap<>();
+    }
+
+    protected List<T> getAllFromMap() {
+        List<T> data = new ArrayList<>(dataMap.values());
+        return data;
+    }
+
+    protected T getById(int id) {
+        return dataMap.get(id);
+    }
+
+    public boolean insert (T t) {
```

Comment:
@DespotOfMorea Since we already have `update` and `insert`, I'd consider allowing insert only if there is no value already (and throw an exception if there already was a value). You can use `map.putIfAbsent()` for this Current semantic of `insert` is "upsert" (insert or update).